tools/unit-tests-merge.cs

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env /Library/Frameworks/Mono.framework/Commands/csharp
+using System.IO;
+using System.Text;
+using System.Xml;
+
+var args = Environment.GetCommandLineArgs ();
+// first arg is "/Library/Frameworks/Mono.framework/Versions/4.8.0/lib/mono/4.5/csharp.exe"
+// second arg the script itself
+// then comes the ones we care about, the path with all the xunit and nunit files
+
+if (args.Length < 3) {
+    Console.WriteLine ($"Missing path for the directory that contains the test results files.");
+    Environment.Exit (-1);
+}
+
+var testsDirectory = args [2];
+if (!Directory.Exists (testsDirectory)) {
+    Console.WriteLine ($"Could not find directory {testsDirectory}");
+    Environment.Exit (-1);
+}
+
+var testFilePattern = "test-*.xml";
+try {
+    var files = Directory.GetFiles (testsDirectory, testFilePattern, SearchOption.AllDirectories);
+    if (files.Length == 0) {
+        Console.WriteLine ($"Not files were found with the pattern '{testFilePattern}'");
+        Environment.Exit (-1);
+    }
+
+    if (files.Length == 1) {
+        // we only have a single tests, so we do not need to do anything
+        Console.WriteLine ("Only one file was found, doing nothing.");
+        Environment.Exit (0);
+    }
+
+    // get the total errors, failures, etc.. from all the files, this will be used to later
+    // write the main test-results node using the addition of all of them
+    var errors = 0L;
+    var inconclusive = 0L;
+    var ignored = 0L;
+    var invalid = 0L;
+    var notRun = 0L;
+    var total = 0L;
+    var failures = 0L;
+    var skipped = 0L;
+    var date = DateTime.Today;
+    TimeSpan? time = null;
```

Comment:
can't you set it to `Zero` and just add to it later ? (skipping `?` and the `HasValue` check in a loop?)